examples/all-clusters-app/all-clusters-common/src/WhmManufacturer.cpp

```diff
@@ -0,0 +1,270 @@
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kBoiler),
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kOther),
+    };
+
+    // The corresponding list of valid headerDemands
+    uint8_t waterHeaterDemandValues[] = {
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kImmersionElement1),
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kImmersionElement2),
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kHeatPump),
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kBoiler),
+        static_cast<uint8_t>(WaterHeaterTypeBitmap::kOther),
+    };
+
+    // Iterate across the valid waterHeaterTypes seeing which heating sources are available based on heaterTypes.
+    // Set the corresponding bit in the heaterDemand bitmap.
+    BitMask<WaterHeaterTypeBitmap> heaterTypes = dg->GetHeaterTypes();
+
+    uint8_t heaterDemandMask = 0;
+    for (uint8_t idx = 0; idx < sizeof(waterHeaterTypeValues) / sizeof(waterHeaterTypeValues[0]); idx++)
+    {
+        // Is this heating source being used?
+        if (heaterTypes.Raw() & waterHeaterTypeValues[idx])
+        {
+            heaterDemandMask |= waterHeaterDemandValues[idx];
+        }
+    }
+
+    return BitMask<WaterHeaterDemandBitmap>(heaterDemandMask);
+}
+
+Status WhmManufacturer::TurnHeatingOn()
+{
+    Status status = Status::Success;
+
+    ChipLogProgress(AppServer, "WhmManufacturer::TurnHeatingOn");
+
+    WaterHeaterManagementDelegate * dg = GetWhmDelegate();
+
+    if (dg->GetBoostState() == BoostStateEnum::kActive)
+    {
+        mBoostActive = true;
+    }
+
+    return status;
+}
+
+Status WhmManufacturer::TurnHeatingOff()
+{
+    Status status = Status::Success;
+
+    ChipLogProgress(AppServer, "WhmManufacturer::TurnHeatingOff");
+
+    if (mBoostActive)
+    {
+        mBoostActive = false;
+    }
+
+    return status;
+}
+
+Status WhmManufacturer::BoostCommandStarted(uint32_t duration, Optional<bool> oneShot, Optional<bool> emergencyBoost,
+                                            Optional<int16_t> temporarySetpoint, Optional<chip::Percent> targetPercentage,
+                                            Optional<chip::Percent> targetReheat)
+{
+    return Status::Success;
+}
+
+Status WhmManufacturer::BoostCommandCancelled()
+{
+    return Status::Success;
+}
+
+void WhmManufacturer::BoostCommandFinished() {}
+
+WaterHeaterManagementDelegate * GetWhmDelegate()
+{
+    WhmManufacturer * mn = GetWhmManufacturer();
+    VerifyOrDieWithMsg(mn != nullptr, AppServer, "WhmManufacturer is null");
+
+    WaterHeaterManagementDelegate * wg = mn->GetWhmDelegate();
+    VerifyOrDieWithMsg(wg != nullptr, AppServer, "WhmDelegate is null");
+
+    return wg;
+}
+
+void SetTestEventTrigger_BasicInstallationTestEvent()
+{
+    WaterHeaterManagementDelegate * dg = GetWhmDelegate();
+
+    // Simulate installation in a 100L tank full of water at 20C, with a target temperature of 60C, in OFF mode
+    dg->SetTankVolume(100);
```

Comment:
Should this save the values so that they can be restored later?

Reply:
See above. I'll do this if needs be....